File: programmers/etc/level2/스킬트리.py
def solution(skill, skill_trees):
    answer = 0
    for s in skill_trees:
        tmp = ""
        for i in s:
            if i in skill:
                tmp += i
        if skill[:len(tmp)] == tmp:
            answer += 1
    return answer


skill = "CBD"
skill_trees = ["BACDE", "CBADF", "AECB", "BDA"]
print(solution(skill, skill_trees)) # 2

File: Java/programmers/etc/level2/스킬트리.java
package CodeTest.Java.programmers.etc.level2;

public class 스킬트리 {
    public int solution(String skill, String[] skill_trees) {
        int answer = 0;
        
        for (String s : skill_trees) {
            String tmp = "";
            for (String i : s.split("")) {
                if (skill.contains(i)) tmp += i;
            }
            if (skill.substring(0, tmp.length()).equals(tmp)) {
                answer++;
            }
        }
        
        return answer;
    }
}
